homeassistant/components/modbus/__init__.py

```diff
@@ -189,6 +191,15 @@
                 CONF_SWAP_WORD_BYTE,
             ]
         ),
+        vol.Optional(CONF_REGISTER_SIZE_BYTES, default=RegisterBytes.NOT_SET): vol.In(
```

Comment:
The default for register_size should be 2.

NOT_SET is a strange default, and should not be allowed as user input.